### PR TITLE
[Lib] Add lib to match pvc, pv, heketi volume, and gluster volume

### PR DESCRIPTION
**Pull request by SushilG96**
This lib will match pvc, pv, heketi volume, gluster volume having the same prefix.

### CODE REVIEW

File: openshift-storage-libs/openshiftstoragelibs/baseclass.py [hunk 1]
```diff
@@ -1133,3 +1137,20 @@ def validate_glusterfsd_memory_usage(self, size_limit):
                     "Failed memory used  of glusterfsd {} is greater than the"
                     " expected size {} for node/pod {}".format(
                         memory_used, size_limit, gluster_node))
+
+    def match_volumes(self, prefix):
```
Comment (vamahaja) [inline openshift-storage-libs/openshiftstoragelibs/baseclass.py]: This function is not specific to scale, move this function to baseclass.

Comment (vamahaja) [inline openshift-storage-libs/openshiftstoragelibs/baseclass.py]: Update function name to `match_volumes_by_prefix`

File: openshift-storage-libs/openshiftstoragelibs/baseclass.py [hunk 2]
```diff
@@ -1133,3 +1137,20 @@ def validate_glusterfsd_memory_usage(self, size_limit):
...
+        """Match pvc, pv, heketi volume, and gluster volume
```
Comment (vamahaja) [inline openshift-storage-libs/openshiftstoragelibs/baseclass.py]: Use below description -
```
Match PVC, PV, Heketi and Gluster volume
```

**Comment by vamahaja** [comment]
Update coommit message to -
```
[Lib] Add baseclass function to match volume name by prefix. 
```

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: openshift-storage-libs/openshiftstoragelibs/baseclass.py (hunk 2 above)
Comment (SushilG96) [reply openshift-storage-libs/openshiftstoragelibs/baseclass.py]: Done

File: openshift-storage-libs/openshiftstoragelibs/baseclass.py (hunk 1 above)
Comment (SushilG96) [reply openshift-storage-libs/openshiftstoragelibs/baseclass.py]: Done

Comment (SushilG96) [reply openshift-storage-libs/openshiftstoragelibs/baseclass.py]: Yes, okay moving to baseclass

**Review by vamahaja — APPROVED** [review]

File: openshift-storage-libs/openshiftstoragelibs/baseclass.py [hunk 3]
```diff
@@ -739,6 +743,23 @@ def validate_file_volumes_count(self, h_node, h_server, node_ip):
...
 
+    def match_volume_by_prefix(self, prefix):
+        """Match PVC, PV, Heketi and Gluster volume
+
+        Args:
+            prefix (str): Start of the unique string.
+        """
+        match_pvc_and_pv(self.h_node, prefix)
+        h_vols = heketi_volume_list_by_name_prefix(
+            self.h_node, self.h_server, prefix, json=True)
```
Comment (adityaramteke) [inline openshift-storage-libs/openshiftstoragelibs/baseclass.py]: I think h_node and h_server are not defined anywhere in baseclass. Better to add them in function parameters.

File: openshift-storage-libs/openshiftstoragelibs/baseclass.py [hunk 4]
```diff
@@ -739,6 +743,23 @@ def validate_file_volumes_count(self, h_node, h_server, node_ip):
             "Failed to verify volume count Expected:'{}', Actual:'{}'".format(
                 len(h_volume_list['volumes']), len(vol_list)))
 
+    def match_volume_by_prefix(self, prefix):
+        """Match PVC, PV, Heketi and Gluster volume
```
Comment (adityaramteke) [inline openshift-storage-libs/openshiftstoragelibs/baseclass.py:747]: Keep the heketi and gluster as in lower case.

**Review by adityaramteke — CHANGES_REQUESTED** [review]

File: openshift-storage-libs/openshiftstoragelibs/baseclass.py (hunk 3 above)
Comment (vamahaja) [reply openshift-storage-libs/openshiftstoragelibs/baseclass.py]: +1

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: openshift-storage-libs/openshiftstoragelibs/baseclass.py (hunk 3 above)
Comment (SushilG96) [reply openshift-storage-libs/openshiftstoragelibs/baseclass.py]: Done

File: openshift-storage-libs/openshiftstoragelibs/baseclass.py (hunk 4 above)
Comment (vamahaja) [reply openshift-storage-libs/openshiftstoragelibs/baseclass.py:747]: @adityaramteke why in lower case ?

**Comment by srivickynesh** [comment]
LGTM

**Review by srivickynesh — APPROVED** [review]

File: openshift-storage-libs/openshiftstoragelibs/baseclass.py (hunk 4 above)
Comment (adityaramteke) [reply openshift-storage-libs/openshiftstoragelibs/baseclass.py:747]: Always lowercase program/project when the word stands alone. Heketi and Gluster are also names of an organization. I agree to keep them in capitalize.

**Review by adityaramteke — APPROVED** [review]

**Review by vamahaja — APPROVED** [review]

**Pull request merged**